scripts/release.sh

```
@@ -0,0 +1,97 @@
+dart format --set-exit-if-changed . || { echo "Error: dart format failed — fix formatting first"; exit 1; }
+flutter analyze || { echo "Error: flutter analyze failed — fix issues first"; exit 1; }
+
+# ── Apply version bump (with rollback on failure) ─────────────────────────────
+
+echo "Bumping: ${CURRENT_VERSION}+${CURRENT_BUILD} → ${NEW_VERSION}+${NEW_BUILD}"
+
+ORIGINAL_VERSION_LINE=$(grep '^version:' "$PUBSPEC")
+rollback() {
+  echo "Rolling back pubspec.yaml..."
+  TMPFILE=$(mktemp)
+  sed "s/^version: .*/$(echo "$ORIGINAL_VERSION_LINE" | sed 's/version: //')/" "$PUBSPEC" > "$TMPFILE"
```

Shellcheck (0.11.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)